### PR TITLE
jpeg2000: enhance parsing of JPEG 2000 codestream main header

### PR DESCRIPTION
**Pull request by bradh**
This updates the existing SIZ parsing, and adds CAP segment parsing.

Also adds unit tests, and an optional test coverage check.

### CODE REVIEW

File: CMakeLists.txt
Comment (kmilos) [inline CMakeLists.txt:1]: Take this out to a sperate PR?

Comment (bradh) [reply CMakeLists.txt:1]: Do you mean the coverage part? That could be dropped if its not wanted. I just found it useful when writing the tests for the parsing logic.

Comment (farindk) [reply CMakeLists.txt:1]: I'll just merge both together... Not necessary to spend the time just to split that.

Comment (kmilos) [reply CMakeLists.txt:1]: Yes, just this CMakeLists.txt change - it's kinda independent of this J2K work, and can be useful on its own, no?

Comment (kmilos) [reply CMakeLists.txt:1]: > I'll just merge both together... Not necessary to spend the time just to split that.

At least separate commits then please. Easier to follow though git blame, eventually revert etc.

**Comment by farindk** [comment]
Thanks, Brad. I've split it into two commits and fixed the uin16_t datatype.

**Pull request closed without merging**